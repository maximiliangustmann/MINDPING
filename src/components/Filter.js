import React from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

Filter.propTypes = {
  filters: PropTypes.arrayOf(PropTypes.string).isRequired,
  active: PropTypes.string,
  activeFilter: PropTypes.func,
}

export default function Filter({ filters, active, activeFilter }) {
  return (
    <FilterBarWrapper>
      <FilterBar data-cy="filterbar">
        {filters.map((filter) => (
          <FilterButton
            key={filter}
            active={active === filter}
            onClick={() => activeFilter(filter)}
          >
            {filter}
          </FilterButton>
        ))}
      </FilterBar>
    </FilterBarWrapper>
  )
}

const FilterBar = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin: 20px;
  padding: 5px;
  border-radius: 50px;
  justify-content: space-evenly;
  align-content: center;
  position: fixed;
  bottom: 0;
  background: var(--gradient-black);
  box-shadow: var(--shadow-green);
  border: 2px solid var(--main-green);
`

const FilterButton = styled.button`
  border: 2px solid var(--main-green);
  background: ${({ active }) =>
    active ? 'var(--main-green)' : 'var(--main-white)'};
  border-radius: 20px;
  margin: 10px 5px;
  color: var(--main-black);
  font-weight: 300;
  font-size: 1em;
`

const FilterBarWrapper = styled.div`
  display: flex;
  @media (min-width: 574px) {
    justify-content: center;
  }
`
