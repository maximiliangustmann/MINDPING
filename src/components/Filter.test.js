import React from 'react'
import renderer from 'react-test-renderer'
import Filter from './Filter'

describe('Filter', () => {
  it('renders correctly', () => {
    const filters = [
      'Show all',
      'Book',
      'Audiobook',
      'Podcast',
      'Film',
      'Series',
    ]
    const tree = renderer.create(<Filter filters={filters} />)
    expect(tree).toMatchSnapshot()
  })
})
