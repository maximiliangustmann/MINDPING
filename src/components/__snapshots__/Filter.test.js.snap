// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter renders correctly 1`] = `
<div
  className="Filter__FilterBar-sc-1lxxwpc-0 eklqMS"
  data-cy="filterbar"
>
  <button
    className="Filter__FilterButton-sc-1lxxwpc-1 iukrbX"
    onClick={[Function]}
  >
    Show all
  </button>
  <button
    className="Filter__FilterButton-sc-1lxxwpc-1 iukrbX"
    onClick={[Function]}
  >
    Book
  </button>
  <button
    className="Filter__FilterButton-sc-1lxxwpc-1 iukrbX"
    onClick={[Function]}
  >
    Audiobook
  </button>
  <button
    className="Filter__FilterButton-sc-1lxxwpc-1 iukrbX"
    onClick={[Function]}
  >
    Podcast
  </button>
  <button
    className="Filter__FilterButton-sc-1lxxwpc-1 iukrbX"
    onClick={[Function]}
  >
    Film
  </button>
  <button
    className="Filter__FilterButton-sc-1lxxwpc-1 iukrbX"
    onClick={[Function]}
  >
    Series
  </button>
</div>
`;
